File: php/features.php
<?php
declare(strict_types=1);

// Tempmail SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class TempmailFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TempmailBaseFeature();
            case "test":
                return new TempmailTestFeature();
            default:
                return new TempmailBaseFeature();
        }
    }
}

File: py/tempmail_sdk/features.py
# Tempmail SDK feature factory

from tempmail_sdk.feature.base_feature import TempmailBaseFeature
from tempmail_sdk.feature.test_feature import TempmailTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TempmailBaseFeature(),
        "test": lambda: TempmailTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
